src/wallet/mod.rs

```diff
@@ -831,12 +847,18 @@ where
         }
 
         if drain_val.is_dust(&drain_output.script_pubkey) {
-            fee_amount += drain_val;
+            // We are not going to add a drain output. This means that both the value
+            // of the output and the fees needed for spending it will go in fees
+            fee_amount += drain_val + drain_fee;
```

Comment:
Is this change the thing that fixes the bug that was found in #660?

Reply:
> Is this change the thing that fixes the bug that was found in https://github.com/bitcoindevkit/bdk/issues/660?

Not really, I thought so, but the bug is still there!

You're right, this specific line of code doesn't seem to make much sense. I'm still trying to figure out what the bug could be - but I'm near! I'll put a comment when I'm ready, for now this PR is not really ready for review :)

Reply:
I updated the PR (and this line changed), now it should be good :)

Reply:
Thanks. The code makes sense now. At least more sense than the code before (that I think I wrote). If you could annotate somehow the change that fixes #660 or explain how it fixed it I'd appreciate it (I know this is still in draft -- take your time).

Reply:
I've added an explanation of the bug in the commit message (da712c45806c663045d6c08f1bb0c99b1b84cd6c), but the TL;DR is that we were using `saturating_sub` for calculating `drain_val`, which means that the `fee_amount` calculation would end up wrong when `drain_val` was supposed to be negative.

I fear that the commit message is not super explicative, so if you have ideas on how to improve it, please let me know!